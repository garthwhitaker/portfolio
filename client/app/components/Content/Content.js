import React from "react";
import GitHubIcon from "@material-ui/icons/GitHub";
import AssigmentIcon from "@material-ui/icons/Assignment";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import Grid from "@material-ui/core/Grid";
import withStyles from "@material-ui/core/styles/withStyles";
import PropTypes from "prop-types";
// eslint-disable-next-line import/no-unresolved
// import { Books } from "../typescript/Books/Books";

const styles = {
  icon: {
    width: "3em",
    height: "3em",
    paddingLeft: 10,
    paddingRight: 10,
  },
  row: {
    display: "flex",
    padding: 20,
  },
  item: {
    fontFamily: '"Open Sans", sans-serif',
    textAlign: "justify",
  },
};

const Content = ({ classes: { row, icon, item } }) => (
  <Grid container>
    <Grid
      container
      alignItems="center"
      direction="row"
      justify="center"
      className={row}
    >
      <Grid item xs={12} sm={4} md={2}>
        <GitHubIcon className={icon} />
      </Grid>
      <Grid item xs={12} sm={8} md={10} className={item}>
        For things I am currently working on or have worked on, check&nbsp;
        <a href="https://github.com/garthwhitaker">GitHub</a>
      </Grid>
    </Grid>
    <Grid
      container
      alignItems="center"
      direction="row"
      justify="center"
      className={row}
    >
      {/* <Grid item xs={12} sm={4} md={2}>
        <AssigmentIcon className={icon} />
      </Grid> */}

      {/* <Grid item xs={12} sm={8} md={10} className={item}>
        To read my latest blog posts, see&nbsp;
        <a href="http://www.garthwhitaker.co.za/blog">Blog</a>
      </Grid> */}
    </Grid>
    <Grid
      container
      alignItems="center"
      direction="row"
      justify="center"
      className={row}
    >
      <Grid item xs={12} sm={4} md={2}>
        <LinkedInIcon className={icon} />
      </Grid>
      <Grid item xs={12} sm={8} md={10} className={item}>
        My professional career in IT can be viewed on&nbsp;
        <a href="https://www.linkedin.com/in/garthwhitaker">LinkedIn</a>
      </Grid>
    </Grid>
    <Grid
      container
      alignItems="flex-start"
      direction="row"
      justify="center"
      className={row}
    >
      {/* <Books /> */}
    </Grid>
  </Grid>
);

Content.propTypes = {
  classes: PropTypes.shape({
    row: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    item: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(Content);
