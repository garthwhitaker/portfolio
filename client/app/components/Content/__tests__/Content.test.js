import React from 'react';
import renderer from 'react-test-renderer';
import Content from './../Content';

describe('Content', () => {
  it('it renders correctly', () => {
    const tree = renderer
      .create(<Content />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
