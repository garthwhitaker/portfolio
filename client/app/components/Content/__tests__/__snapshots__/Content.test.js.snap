// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content it renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root Content-row-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Content-icon-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 18c1.1 0 1.99-.9 1.99-2L22 5c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2H0c0 1.1.9 2 2 2h20c1.1 0 2-.9 2-2h-4zM4 5h16v11H4V5zm8 14c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root Content-item-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-10"
    >
      For things I am currently working on or have worked on, check 
      <a
        href="https://github.com/garthwhitaker"
      >
        GitHub
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-root Content-row-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Content-icon-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root Content-item-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-10"
    >
      To read my latest blog posts, see 
      <a
        href="http://www.garthwhitaker.co.za/blog"
      >
        Blog
      </a>
    </div>
  </div>
  <div
    className="MuiGrid-root Content-row-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Content-icon-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root Content-item-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-10"
    >
      My professional career in IT can be viewed on 
      <a
        href="https://www.linkedin.com/in/garthwhitaker"
      >
        LinkedIn
      </a>
    </div>
  </div>
</div>
`;
